app/services/albums.js

```diff
@@ -28,3 +29,38 @@ exports.listAlbumPhotos = albumId =>
     logger.error(util.inspect(err));
     throw apiAlbumsError(err.message);
   });
+
+exports.getAlbumData = async albumId => {
+  try {
+    const albumDataResponse = await rp({
+      uri: `${url}/albums/${albumId}`,
+      resolveWithFullResponse: true,
+      json: true
+    });
+    return albumDataResponse.body;
+  } catch (err) {
+    logger.error(util.inspect(err));
+    throw apiAlbumsError('Album not bought: could not find album');
```

Comment:
You say me, "When the album doesn't exist, the status code is a 200 with an empty body". So, If the code enters in the catch block, means that the API's response is different to 200. So, if really because the album doesn't exist?